### PR TITLE
Add block timestamp to block header

### LINKED ISSUE #241: Get ready for adding the timestamp field to the block header
https://github.com/bpfkorea/agora/pull/1437/files will add an 8-byte timestamp to the block header.

```diff
diff --git a/source/agora/consensus/data/Block.d b/source/agora/consensus/data/Block.d
index 33f218b22..4eb934a9b 100644
--- a/source/agora/consensus/data/Block.d
+++ b/source/agora/consensus/data/Block.d
@@ -65,6 +65,9 @@ public struct BlockHeader
     /// List of indices to the validator UTXO set which have not revealed the preimage
     public uint[] missing_validators;

+    /// Block unix timestamp
+    public ulong timestamp;
+
```

It would be good to prepare a PR with the changes in Stoa.

### PR DESCRIPTION
**Pull request by TrustHenry**
Upgrade SDK to 0.0.13
Tests changes and due to the addition of type timestamp in the block header.

Related to #241

### CODE REVIEW

File: src/modules/storage/LedgerStorage.ts [hunk 1]
```diff
@@ -1130,7 +1130,7 @@ export class LedgerStorage extends Storages
                 SELECT
                     TX.address,
                     TX.block_height as height,
-                    STRFTIME('%s', '2020-01-01 00:00:00') * 1000 + TX.block_height * 10 * 60000 as block_time,
+                    TX.time_stamp as block_time,
```
Comment (MichaelKim20) [inline src/modules/storage/LedgerStorage.ts:1133]: `unlock_time` also needs to be changed.
```
    (TX.time_stamp + (TX.unlock_height - TX.block_height) * 10 * 60) as unlock_time,
```

File: src/modules/storage/LedgerStorage.ts [hunk 2]
```diff
@@ -1228,11 +1230,10 @@ export class LedgerStorage extends Storages
     {
         let hash = new Hash(tx_hash).toBinary(Endian.Little);
 
-        // TODO We should apply the block's timestamp to this method when it is added
         let sql_tx =
             `SELECT
                 T.block_height as height,
-                STRFTIME('%s', '2020-01-01 00:00:00') * 1000 + T.block_height * 10 * 60000 as block_time,
+                B.time_stamp as block_time,
```
Comment (MichaelKim20) [inline src/modules/storage/LedgerStorage.ts:1236]: `unlock_time` also needs to be changed.
```
  (B.time_stamp + (T.unlock_height - T.block_height) * 10 * 60) as unlock_time,
```

File: package-lock.json [hunk 3]
```diff
@@ -3,6 +3,3704 @@
   "version": "0.0.0",
   "lockfileVersion": 1,
   "requires": true,
+  "packages": {
```
Comment (MichaelKim20) [inline package-lock.json]: It seems that the change in package-lock.json is more than necessary.
Delete both the npm-modules folder and package-lock.json, then run npm install

File: tests/data/Block.0.sample1.json [hunk 4]
```diff
@@ -1,126 +1,130 @@
...
-                    "address": "GCOQEOHAUFYUAC6G22FJ3GZRNLGVCCLESEJ2AXBIJ5BJNUVTAERPLRIJ"
-                },
-                {
-                    "value": "610000000000000",
-                    "address": "GCOQEOHAUFYUAC6G22FJ3GZRNLGVCCLESEJ2AXBIJ5BJNUVTAERPLRIJ"
-                }
-            ],
-            "payload": ""
-        }
+    "header":{
```
Comment (MichaelKim20) [inline tests/data/Block.0.sample1.json]: Please print out JSON Pretty and make it easier to compare with previous data.
https://jsonformatter.org/json-pretty-print

File: tests/data/Block.2.sample1.json [hunk 5]
```diff
@@ -1,71 +1,74 @@
...
-    "header": {
-        "prev_block": "0x2ba067da82aa7a02c49771cc07146d786b3840585c6125d79e4b4914c62347a4bc900c8b83dc369d0d3a45493549086926ada889d532ccde4b9ee2e3d12d4844",
-        "height": "2",
-        "merkle_root": "0x5ed0db3335189a96f23b00787bc4e34d7f0c20ab9536f3e951c9049299e5f2509656f858fc0a6ed1851948935f7a19bbce9170e3ae767129aa017bdd3001dfc5",
-        "validators": "[]",
-        "signature": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
-        "enrollments": [],
-        "random_seed": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
-        "missing_validators": []
+    "header":{
```
Comment (MichaelKim20) [inline tests/data/Block.2.sample1.json]: Please print out JSON Pretty and make it easier to compare with previous data.
https://jsonformatter.org/json-pretty-print

File: tests/RecoveryData.test.ts [hunk 6]
```diff
@@ -12,1561 +12,1594 @@
...
-                    }
-                ],
-                "payload": ""
-            }
-        ],
-        "merkle_tree": [
-            "0x6314ce9bc41a7f5b98309c3a3d824647d7613b714c4e3ddbc1c5e9ae46db29715c83127ce259a3851363bff36af2e1e9a51dfa15c36a77c9f8eba6826ff975bc",
-            "0x7a5bfeb96f9caefa377cb9a7ffe3ea3dd59ea84d4a1c66304ab8c307a4f47706fe0aec2a73ce2b186a9f45641620995f8c7e4c157cee7940872d96d9b2f0f95c",
-            "0x788c159d62b565655d9f725786c38e6802038ee73d7a9d187b3be1c7de95aa0ba856bf81bb556d7448488e71f4b89ce6eba319d0536798308112416413289254"
+    [
```
Comment (MichaelKim20) [inline tests/RecoveryData.test.ts]: Please print out JSON Pretty and make it easier to compare with previous data.
https://jsonformatter.org/json-pretty-print

File: tests/data/Block.1.sample1.json [hunk 7]
```diff
@@ -1,927 +1,930 @@
...
-                    "address": "GDNB22W7MUI36RFJFGBK6VDYQ23XOMPJYWE53LB6UZH4AYV6RUN76RC4"
-                },
-                {
-                    "value": "24400000000000",
-                    "address": "GDNC22KXQGDKH6EO2UIOWNCYOPMO6XZX7T6P4TH6XZ4UYSR6JOBGY5U6"
-                }
-            ],
-            "payload": ""
-        }
+    "header":{
```
Comment (MichaelKim20) [inline tests/data/Block.1.sample1.json]: Please print out JSON Pretty and make it easier to compare with previous data.
https://jsonformatter.org/json-pretty-print

**Review by MichaelKim20 — CHANGES_REQUESTED** [review]
There are three modifications that need to be made. The rest is very good.
1. The JSON data needs to be sorted.
2. Calculate the predicted value of unlock_time.
3. Check package-lock.json again.

File: src/modules/storage/LedgerStorage.ts (hunk 1 above)
Comment (TrustHenry) [reply src/modules/storage/LedgerStorage.ts:1133]: https://github.com/bpfkorea/stoa/issues/252

File: package-lock.json (hunk 3 above)
Comment (MichaelKim20) [reply package-lock.json]: Since the value of 'lockfileVersion' is being forced to 1 by 'Bump PR', we would like to use version 14.XX of NodeJS.

**Comment by TrustHenry** [comment]
@MichaelKim20 Thanks for the reviews. I apply the suggestions.

File: package-lock.json [hunk 8]
```diff
@@ -46,20 +47,18 @@
...
     "@types/cors": {
-      "version": "2.8.8",
-      "resolved": "https://registry.npmjs.org/@types/cors/-/cors-2.8.8.tgz",
-      "integrity": "sha512-fO3gf3DxU2Trcbr75O7obVndW/X5k8rJNZkLXlQWStTHhP71PkRqjwPIEI0yMnJdg9R9OasjU+Bsr+Hr1xy/0w==",
-      "requires": {
-        "@types/express": "*"
-      }
+      "version": "2.8.9",
+      "resolved": "https://registry.npmjs.org/@types/cors/-/cors-2.8.9.tgz",
+      "integrity": "sha512-zurD1ibz21BRlAOIKP8yhrxlqKx6L9VCwkB5kMiP6nZAhoF5MvC7qS1qPA7nRcr1GJolfkQC7/EAL4hdYejLtg=="
```
Comment (TrustHenry) [inline package-lock.json:61]: upgraded @types/cors 2.8.9

File: src/Stoa.ts [hunk 9]
```diff
@@ -117,8 +117,6 @@ class Stoa extends WebService
         // parse application/json
         this.app.use(bodyParser.json())
         this.app.use(cors(cors_options));
-        // enable pre-flight
-        this.app.options('*', cors(cors_options));
```
Comment (TrustHenry) [inline src/Stoa.ts:121]: This phrase is obsolete.

**Pull request closed without merging**